quickcheck-threatmodel/src/Test/QuickCheck/ThreatModel/TxModifier.hs

```diff
@@ -196,8 +249,44 @@ applyTxMod tx utxos (AddInput addr value datum) =
 
     scriptData' = recomputeScriptData Nothing idxUpdate scriptData
 
-applyTxMod tx utxos (AddPlutusScriptInput script value datum redeemer) =
-    ( Tx (ShelleyTxBody era body{Ledger.btbInputs = inputs'} scripts' scriptData' auxData validity) wits
+applyTxMod tx utxos (AddInput addr value datum rscript True) =
+    ( Tx (ShelleyTxBody era body{Ledger.btbReferenceInputs = refInputs} scripts scriptData auxData validity) wits
+    , utxos' )
+  where
+    Tx (ShelleyTxBody era body@Ledger.BabbageTxBody{..} scripts scriptData auxData validity) wits = tx
+
+    txIn    = TxIn dummyTxId (TxIx txIx)
+    txIx    = maximum $ 0 : map (+ 1) [ ix | TxIn txId (TxIx ix) <- Map.keys $ unUTxO utxos, txId == dummyTxId ]
```

Comment:
This is happening a few times, and seems to be the only use of `dummyTxId`. It's also not very clear what's going on, so it might be good to turn this into a function with a descriptive name.